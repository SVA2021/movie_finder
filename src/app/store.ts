import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import authReducer from '../features/auth/authSlice';
import movieFinderReducer from '../features/movieFinder/movieFinderSlice';

export const store = configureStore({
  reducer: {
    auth: authReducer,
		movieFinder: movieFinderReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
