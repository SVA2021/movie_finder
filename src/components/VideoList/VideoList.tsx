import {Box, useTheme} from '@mui/material';
import {FC, memo, useEffect, useRef} from 'react';
import {useParams} from 'react-router-dom';
import {Swiper, SwiperSlide} from 'swiper/react';
import 'swiper/swiper.min.css'; // core Swiper
import {EmptyDataBox, VideoItem} from '..';
import {useAppDispatch, useAppSelector} from '../../app/hooks';
import {selectDetails} from '../../features/movieFinder/movieFinderSlice';
import {getMovieVideosAsync} from '../../features/movieFinder/movieFinderThunks';

export const VideoList: FC = memo(() => {

  const dispatch = useAppDispatch();
  const DetailsPage = useAppSelector(selectDetails);
  const MovieTrailers = DetailsPage.videos?.items ?? [];
  const theme = useTheme();
  const swiperRef = useRef<any>();
  const params = useParams();
  const id = Number(params.id)

  useEffect(() => {
    dispatch(getMovieVideosAsync(id))
  }, [id]);

  return (
    <Box data-testid={'video-list'}>
      {MovieTrailers.length > 0
        ? <Swiper
          onSwiper={(swiper) => swiperRef.current = swiper}
          slidesPerView={1}
          spaceBetween={8}
          breakpoints={
            {
              [theme.breakpoints.values.md]: {
                slidesPerView: 3,
                spaceBetween: 16,
              },
              [theme.breakpoints.values.lg]: {
                slidesPerView: 5,
                spaceBetween: 24,
              },
            }
          }
        >
          {
            MovieTrailers.map((item, index) =>
              <SwiperSlide key={index} >
                <VideoItem item={item} />
              </SwiperSlide>
            )
          }
        </Swiper>
        : <EmptyDataBox />
      }
    </Box>
  );
});