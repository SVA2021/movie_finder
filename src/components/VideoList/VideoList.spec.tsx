import {render, screen} from "@testing-library/react";
import {Provider} from 'react-redux';
import {BrowserRouter} from 'react-router-dom';
import {store} from "../../app/store";
import {VideoList} from "./VideoList";

describe('render VideoList', () => {

  const setUp = () => render(
    <Provider store={store}>
      <BrowserRouter>
        <VideoList />
      </BrowserRouter>
    </Provider>
  );

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('video-list')).toMatchSnapshot();
  });

  it('check empty text', () => {
    setUp();
    expect(screen.getByText(/нет данных/i)).toBeInTheDocument();
  });

})