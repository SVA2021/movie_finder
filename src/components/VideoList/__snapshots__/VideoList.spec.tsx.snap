// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render VideoList snapshot test 1`] = `
<div
  class="MuiBox-root css-0"
  data-testid="video-list"
>
  <p
    class="MuiTypography-root MuiTypography-h3 css-13mkr37-MuiTypography-root"
    data-testid="empty-data-box"
  >
    Нет данных
  </p>
</div>
`;
